packages/EasySecurity/src/Bridge/Symfony/Security/RoleExpressionFunctionProvider.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasySecurity\Bridge\Symfony\Security;
+
+use EonX\EasySecurity\Bridge\Symfony\Exceptions\RoleConstantNotFoundException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\ExpressionLanguage\ExpressionFunction;
+use Symfony\Component\ExpressionLanguage\ExpressionFunctionProviderInterface;
+
+final class RoleExpressionFunctionProvider implements ExpressionFunctionProviderInterface
+{
+    /**
+     * @var mixed[]
+     */
+    private array $cached = [];
```

Comment:
I don't see who sets that. Do we need it?

Reply:
See below on the line 48 (`$this->cached[$role] = \constant($constant);`).